.github/workflows/cleanup.yml

```diff
@@ -1,14 +1,38 @@
 name: cleanup
 
 on:
-  schedule:
-    - cron: '*/5 * * * *'
+  # Triggerred on every workflow run except on master
+  workflow_run:
+    workflows: [ "ci" ]
+    types:
+      - requested
+    branches-ignore: [ "master" ]
+  # Triggerred on every PR close
+  pull_request:
+    types: [ "closed" ]
 
 jobs:
   cancel:
     runs-on: ubuntu-latest
+    timeout-minutes: 5
     steps:
-      - uses: n1hility/cancel-previous-runs@2e3c1893986568a2197c41957b9c809cbcf1a61e
+      - name: 'Cancel All Runs Except Latest'
+        # For pull_request events different config is needed
+        if: ${{ github.event_name != 'pull_request' }}
```

Comment:
Maybe this should be two separate workflow files?

Reply:
It was leftover - see the new squashed state.